### PR TITLE
Use an arena in ContinuousTrajectory

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: physics/continuous_trajectory.hpp [hunk 1]
```diff
@@ -140,27 +142,26 @@ class ContinuousTrajectory : public Trajectory<Frame> {
...
     Instant t_max;
-    not_null<std::unique_ptr<Polynomial<Displacement<Frame>, Instant>>>
-        polynomial;
+    not_null<Polynomial<Displacement<Frame>, Instant>*> polynomial;
   };
-  using InstantPolynomialPairs = std::vector<InstantPolynomialPair>;
+  using InstantPolynomialPairs =
+      std::vector<InstantPolynomialPair, ArenaAllocator<InstantPolynomialPair>>;
 
   // May be overridden for testing.
```
Comment (eggrobin) [inline physics/continuous_trajectory.hpp]: Comment: the result is owned by `arena_`.

File: physics/continuous_trajectory.hpp [hunk 2]
```diff
@@ -140,27 +142,26 @@ class ContinuousTrajectory : public Trajectory<Frame> {
   // stored in this vector sorted by their |t_max|, as it turns out that we
   // never need to extract their |t_min|.  Logically, the |t_min| for a
   // polynomial is the |t_max| of the previous one.  The first polynomial has a
-  // |t_min| which is |*first_time_|.
+  // |t_min| which is |*first_time_|.  The pointer is owned by |arena_|;
```
Comment (eggrobin) [inline physics/continuous_trajectory.hpp]: `*polynomial` is owned by `arena_` (so is `polynomial` in that this struct is allocated there by the vector, but that's not the intent of that comment).

**Comment by eggrobin** [comment]
This means `ForgetBefore` will not reduce memory usage in-game until the next scene change, which might be a problem.

**Comment by pleroy** [comment]
Closing this PR.  We'll use `std::pmr` once it's real.

**Pull request closed without merging**